src/couch_replicator.erl

```diff
@@ -129,3 +135,53 @@ cancel_replication(RepId, #user_ctx{name = Name, roles = Roles}) ->
             {error, not_found}
         end
      end.
+
+
+-spec stream_terminal_docs_info(binary(), user_doc_cb(), any()) -> any().
+stream_terminal_docs_info(Db, Cb, UserAcc) ->
+    DDoc = <<"_replicator">>,
+    View = <<"terminal_states">>,
+    QueryCb = fun handle_replicator_doc_query/2,
+    Args = #mrargs{view_type = map, reduce = false},
+    Acc = {Db, Cb, UserAcc},
+    try fabric:query_view(Db, DDoc, View, QueryCb, Acc, Args) of
+    {ok, {Db, Cb, UserAcc1}} ->
+        UserAcc1;
+    {not_found, missing} ->
+        UserAcc
+    catch error:database_does_not_exist ->
+        UserAcc
```

Comment:
Main reason is because response are streamed back to the user and the response has already been sent as `chttpd:start_delayed_json_response(Req, 200, [], "["),` by the time this is called. The idea was that if database is not created it is not a fatal error, the callback is never called and we just return the UserAcc back to them (which will actually be the Response object), so user (showroom module) can continue to pass it along to other stream_* functions.